src/components/learner-credit-management/tests/BudgetDetailPage.test.jsx

```diff
@@ -2359,20 +2359,18 @@ describe('<BudgetDetailPage />', () => {
     if (field === 'status') {
       const filtersButton = getButtonElement('Filters', { screenOverride: assignedSection });
       await user.click(filtersButton);
-      await waitFor(async () => {
-        const filtersDropdown = screen.getAllByRole('group', { name: 'Status' })[0];
-        const filtersDropdownContainer = within(filtersDropdown);
-        if (value.includes('waiting')) {
-          const waitingForLearnerOption = filtersDropdownContainer.getByLabelText('Waiting for learner 1', { exact: false });
-          expect(waitingForLearnerOption).toBeInTheDocument();
-          await user.click(waitingForLearnerOption);
-
-          expect(waitingForLearnerOption).toBeChecked();
+      if (value.includes('waiting')) {
+        const waitingForLearnerOption = await screen.findByLabelText(/Waiting for learner/i);
+        expect(waitingForLearnerOption).toBeInTheDocument();
+        waitingForLearnerOption.checked = true;
+        expect(waitingForLearnerOption).toBeChecked();
+        waitingForLearnerOption.click();
```

Comment:
Should click before doing the check, and I'm not seeing a need for `waitingForLearnerOption.checked = true;`
```suggestion
        waitingForLearnerOption.click();
        expect(waitingForLearnerOption).toBeChecked();
```